index.js

```diff
@@ -0,0 +1,164 @@
+const express = require("express");
+const fs = require("fs");
+const app = express();
+
+app.use(express.json());
+
+app.get("/person/:name", (req, res) => {
+  const idadeRandom = Math.floor(Math.random() * (80 - 18) + 18);
+
+  const jsonResponse = JSON.parse(`{
+        "person": {
+            "name": "${req.params.name}",
+            "idade": "${idadeRandom}"
+        } 
+    }`);
+  res.send(jsonResponse);
+});
+
+let persons = [];
+
+(() => {
+  fs.readFile("./persons.txt", (err, data) => {
+    if (data !== undefined) {
+      persons = JSON.parse(data);
+      if (err) {
+        console.log("Cannot read file");
+        console.log(err);
+      }
+    } else {
+      console.log(`persons.txt is empty`);
+    }
+  });
+})();
+
+function savePersons() {
+  fs.writeFile("./persons.txt", JSON.stringify(persons), "utf-8", (err) => {
+    console.log(err);
+  });
+}
+
+app.post("/persons", (req, res) => {
+  const person = req.body;
+  const isPersonEmpty = person.length === 0;
+  const regex = /^[A-Za-z]+$/;
+  personNameLength = person.name.length;
+  personHeight = person.height;
+
+  if (!isPersonEmpty && regex.test(person.name)) {
+    if (personNameLength < 4) {
+      res.status(400).send(`the name is too short(minimum 4 characters)`);
+      res.end();
+      return;
+    } else if (personNameLength > 50) {
+      res.status(400).send(`the name is too long(maximum 50 characters)`);
+      res.end();
+      return;
+    }
+
+    if (personHeight < 50) {
+      // check height of person
+      res.status(400).send(`the height is too low(minimum 50)`);
+      res.end();
+      return;
+    } else if (personHeight > 250) {
+      res.status(400).send(`the height is too high(maximum 250)`);
+      res.end();
+      return;
+    }
+    persons.push(person);
+    res
+      .status(200)
+      .send(`person <strong>${person.name}</strong> was created successfully`);
+    savePersons();
+  } else if (!isPersonEmpty) {
+    res.status(400).send(`your name '${person.name}' should only have letters`);
+  } else {
+    res.status(400);
+  }
+  res.end();
+});
+
+app.get("/persons", (req, res) => {
+  const isPersonsEmpty = persons.length === 0;
+  if (isPersonsEmpty) {
+    res.status(200).send(`<h1>404</h1><p>there's <strong>no one</strong>`);
```

Comment:
Normally when you request a list and it has no items you return an empty array. The way you did is ok, but this is the most common way of returning the response in this situations.

Read about "rest api style" if you want, but we will exercise this later too ;)

Reply:
204 No Content is the appropriate, right?

Reply:
Actually it's 200, with an empty array being returned, because the empty array is the return, and it exists. You should return 404, for example, if someone calling an unknown endpoint, because then the server can't find what is being queried.